include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp

```diff
@@ -321,6 +338,9 @@ SpatialOperator<dim, Number>::evaluate(BlockVectorType &       dst,
   if(param.right_hand_side)
     rhs_operator.evaluate_add(dst.block(block_index_pressure), time);
 
+  if(integrated_rhs_set)
```

Comment:
Here I wonder whether we should be more explicit in order to improve understanding and also to avoid potential mistakes). I think there should be a parameter in `Acoustics::Parameters` (probably section on mathematical model) describing whether the acoustic solver receives a source term from the fluid problem. If this parameter is `true`, I would then check inside the body of the if-statement that the pointer is valid. Outside this class, I think one could do everything with this vector/pointer, so having the state `integrated_rhs_set = true` inside this class (just because the function `set_integrated_rhs()` has been set once) might not imply that this pointer is valid. Put differently, the meaning of the variable used currently is `integrated_rhs_has_been_set_once` in my opinion.

The current implementation looks very generic (`integrated_rhs` can be anything, must not come from a fluid solver), but is this really the case? What if I want to add another source term in the future. I would need some logic outside this class that adds the two vectors together and then sets the vector once. many mistakes are possible: (i) the vectors are not added, (ii) different modules calls `set_integrated_rhs()` independently and overwrite each other. Using a more modular design with separate parameters as suggested above is easier to understand and less error prone in my opinion. Put differently, I do not consider an "explicit dependency/reference" of the acoustics module on/to the fluid module (because of a parameter like `acoustic_param.use_source_term_xyz`) a disadvantage.

Reply:
I resolved this in two commits:
[1d93518](https://github.com/exadg/exadg/pull/635/commits/1d9351801f6339023f4ba84e2507634407dc4921)
[cabe4ae](https://github.com/exadg/exadg/pull/635/commits/cabe4aebb84ce3048032eee61e142eaa49eef43f)

Reply:
There is one problem remaining: A `lazy_ptr` can never be a `nullptr`. It is initially the address to a default constructed object. Hence, we can not check if `lazy_ptr` is set by checking if it is a `nullptr`.

Reply:
I think the idea of the `lazy_ptr` is that it can either be data or a pointer to data. In the present case, it seems to be used always in the sense of a pointer to external data? In that case, it would be reasonable to use a raw pointer (or shared_ptr) in my opinion instead of the `lazy_ptr`.

Reply:
In that case, lets keep the raw pointer :)